### ISSUE TITLE
Can't use search with Japanese

### ISSUE THREAD

**Issue by azumukupoe** [Sandmann79/xbmc#398]
**Addon used**
- [x] Amazon VOD (plugin.video.amazon-test)
- [ ] Amazon (plugin.video.amazon)
- [ ] Browser Launcher (plugin.program.browser.launcher)
- Addon Version 0.8.1

**Prime Video Account (select one)**
- [ ] primevideo.com
- [x] amazon.(com/co.uk/de/jp)

**System Setup (please provide the following information):**
- Hardware: Desktop PC
- OS Version: Windows 10 Pro
- Kodi Version 18.5

**Upload Logs**
[kodi.log](https://github.com/Sandmann79/xbmc/files/4074933/kodi.log)

**Describe the bug**
I don't know if non-English language is supported but at least doesn't work for me.
I add videos to the watchlist as a workaround.

**Comment by Varstahl** [comment]
There are a lot of 504 errors (Gateway timeout) in the logs, about 3200, breaking results in a lot of places including /cdp/ endpoints:
```
[Amazon VOD] Error reason: 504 error (TryAgain)
```
Also, @Sandmann79, there seems to be some unicode problems in the Search function:
```
NOTICE: [Amazon VOD] Version: 0.8.1
NOTICE: [Amazon VOD] Unicode filename support: True
NOTICE: [Amazon VOD] Locale: ja-jp / Language: en
 ERROR: EXCEPTION Thrown (PythonToCppException) : -->Python callback/script returned the following error<--
            - NOTE: IGNORING THIS CAN LEAD TO MEMORY LEAKS!
        Error Type: <type 'exceptions.KeyError'>
        Error Contents: u'\u304b'
        Traceback (most recent call last):
            File "C:\Users\azumu\AppData\Roaming\Kodi\addons\plugin.video.amazon-test\default.py", line 5, in <module>
            EntryPoint()
            File "C:\Users\azumu\AppData\Roaming\Kodi\addons\plugin.video.amazon-test\resources\lib\startup.py", line 99, in EntryPoint
            exec('g.amz.{}()'.format(mode))
            File "<string>", line 1, in <module>
            File "C:\Users\azumu\AppData\Roaming\Kodi\addons\plugin.video.amazon-test\resources\lib\amazontld.py", line 178, in Search
            url = 'searchString=%s%s' % (quote_plus(searchString), self._s.OfferGroup)
            File "C:\Program Files\Kodi\system\python\Lib\urllib.py", line 1306, in quote_plus
            return quote(s, safe)
            File "C:\Program Files\Kodi\system\python\Lib\urllib.py", line 1299, in quote
            return ''.join(map(quoter, s))
        KeyError: u'\u304b'
        -->End of Python script error report<--
```

**Comment by Sandmann79** [comment]
Search should be fixed with b48607e8bc20514ab729522f428db0eb24aa101b. 
The 504 errors seems to be a temporary server or connection problem - checked the links and they are returning results for me.

**Comment by Sandmann79** [comment]
Amazon VOD 0.8.2 released, which fixed this issue. Closing.
